src/js/common/containers/DowntimeNotification.jsx

```diff
@@ -0,0 +1,248 @@
+  // downtime information (each downtime corresponding to a dependency/service name) as the values.
+  calculateDowntime(dependencies, scheduledDowntime) {
+    const now = moment();
+    const nextHour = moment().add(1, 'hour');
+    const downtimeMap = {};
+
+    // Create a map with the service statuses as keys { ok, down, downApproaching }
+    Object.keys(serviceStatus).forEach((status) => { downtimeMap[status] = []; });
+
+    return dependencies
+
+      // Get the corresponding downtime value by the name of the service/dependency
+      .map((serviceName) => scheduledDowntime.find(downtime => downtime.service === serviceName))
+
+      // Remove null values (services that have no known downtime)
+      .filter(downtime => !!downtime)
+
+      // Put each value into the corresponding status of the downtimeMap
+      .reduce((map, downtime) => {
+        const status = this.getStatusForDowntime(downtime, now, nextHour);
+        return {
+          ...map,
+          [status]: map[status].concat(downtime)
+        };
+      }, downtimeMap);
+  }
+
+  determineStatus(downtimeMap) {
+    if (this.props.determineStatus) return this.props.determineStatus(downtimeMap);
+
+    const statusDown = downtimeMap[serviceStatus.down].length > 0;
+    const statusDownApproaching = downtimeMap[serviceStatus.downtimeApproaching].length > 0;
+    const statusOk = !statusDown && !statusDownApproaching;
+
+    if (statusOk) return serviceStatus.ok;
+    if (statusDown) return serviceStatus.down;
+
+    return serviceStatus.downtimeApproaching;
+  }
+
+  renderStatusDown({ endTime }) {
+    const title = `The ${this.props.appTitle} is down for maintenance.`;
+    let message = <p>We’re making some updates to {this.props.appTitle}. We’re sorry it’s not working right now. Please check back soon.</p>;
+    if (endTime) {
+      message = (
+        <p>We’re making some updates to {this.props.appTitle}. We’re sorry it’s not working right now, and we hope to be finished by {endTime.format('MMMM Mo, LT')}. Please check back soon.</p>
+      );
+    }
+
+    let downtimeNotification = null;
+
+    // @todo remove the "true" part of this
+    if (true || this.props.userIsAuthenticated) {
+      downtimeNotification = <div><h2>{title}</h2>{message}</div>;
+    } else {
+      downtimeNotification = <AlertBox isVisible status="warning" headline={<h4>{title}</h4>} content={message}/>;
+    }
+    return (
+      <DowntimeNotificationWrapper status={serviceStatus.down}>
+        {downtimeNotification}
+      </DowntimeNotificationWrapper>
+    );
+  }
+
+  renderStatusDownApproaching({ startTime, endTime }) {
+    const title = `The ${this.props.appTitle} will be down for maintenance soon`;
+    const message = <p>We'll be doing some work on {this.props.appTitle} on {startTime.format('MMMM Mo')} between {startTime.format('LT')} and {endTime.format('LT')} . If you have trouble using this tool during that time, please check back soon.</p>;
```

Comment:
Couple things:
1. Straight quote needs to be a curly one (I'm not sure how to do this on this linux machine, so no example, sorry!)
2. Looks like there's an extra space between `{endTime.format('LT')}` and the following `.`